framework/src/org/checkerframework/framework/type/DefaultTypeHierarchy.java

```diff
@@ -920,6 +920,15 @@ public Boolean visitWildcard_Array(
     @Override
     public Boolean visitWildcard_Declared(
             AnnotatedWildcardType subtype, AnnotatedDeclaredType supertype, VisitHistory visited) {
+        if (subtype.isUninferredTypeArgument() && supertype.getTypeArguments().isEmpty()) {
```

Comment:
If the wildcard is `? extends @Nullable Object` and the supertype is `@Nullable String`, then it is safe to return true.  However if the supertype is `@NullableList<@NonNull String>` then it's not possible to decide if it is a subtype of the wildcard.  Does that make sense?  I'll think about how to clarify the comment.